redundant-haproxy-floatingip-ubuntu/azuredeploy.json

```diff
@@ -0,0 +1,507 @@
+      "type": "string",
+      "defaultValue": "14.04.2-LTS",
+      "allowedValues": [
+        "12.04.5-LTS",
+        "14.04.2-LTS",
+        "15.04"
+      ],
+      "metadata": {
+        "description": "The Ubuntu version for the VM. This will pick a fully patched image of this given Ubuntu version. Allowed values: 12.04.5-LTS, 14.04.2-LTS, 15.04."
+      }
+    },
+    "vmSize": {
+      "type": "string",
+      "defaultValue": "Standard_D1",
+      "metadata": {
+        "description": "Size of the VM"
+      }
+    }
+  },
+  "variables": { 
+    "scriptsBaseUrl": "https://raw.githubusercontent.com/sivaedupuganti/azure-quickstart-templates/master/floatingip-haproxy-keepalived-ubuntu/",
+
+    "numberOfHAproxyInstances": 2,
+    "masterHAproxyInstanceIndex": 0,
+    "haproxyVMScripts": {
+    "fileUris": [
+      "[concat(variables('scriptsBaseUrl'), 'haproxyvm-configure.sh')]",
+      "[concat(variables('scriptsBaseUrl'), 'keepalived-action.sh')]",
+      "[concat(variables('scriptsBaseUrl'), 'keepalived-check-appsvc.sh')]"
+    ],
+    "commandToExecute": "[concat('sudo bash -x haproxyvm-configure.sh ', ' -a ', parameters('appVMNamePrefix'), '0 -a ', parameters('appVMNamePrefix'), '1 -p ', variables('appVMPort'), ' -l ', parameters('dnsNameforLBIP'), '.', resourceGroup().location, '.cloudapp.azure.com -t ', variables('lbVIPPort'), ' -m ', parameters('haproxyVMNamePrefix'), '0 -b ', parameters('haproxyVMNamePrefix'), '1')]"
+  },
+
+    "numberOfAppInstances": 2,
+    "appVMScripts": {
+      "fileUris": [
+        "[concat(variables('scriptsBaseUrl'), 'apache-setup.sh')]"
+      ],
+      "commandToExecute": "sudo bash apache-setup.sh"
+    },
+    "appVMPort": 80,
+
+    "imagePublisher": "Canonical",
+    "imageOffer": "UbuntuServer",
+    "sshKeyPath": "[concat('/home/',parameters('adminUsername'),'/.ssh/authorized_keys')]",
+    "nicNamePrefix": "nic-",
+
+    "storageAccountType": "Standard_LRS",
+    "haproxyAvailabilitySetName": "haproxyAvSet",
+    "appAvailabilitySetName": "appAvSet",
+
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',parameters('vnetName'))]",
+    "addressPrefix": "10.0.0.0/16",
+    "subnetName": "Subnet-1",
+    "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables ('subnetName'))]",
+    "subnetPrefix": "10.0.0.0/24",
+
+    "lbPublicIPAddressType": "Static",
+    "lbPublicIPAddressName": "[concat(parameters('lbName'), '-publicip')]",
+    "lbVIPPort": 80,
+    "lbID": "[resourceId('Microsoft.Network/loadBalancers',parameters('lbName'))]",
+    "frontEndIPConfigID": "[concat(variables('lbID'),'/frontendIPConfigurations/LoadBalancerFrontEnd')]",
+    "lbPoolID": "[concat(variables('lbID'),'/backendAddressPools/BackendPool1')]",
+    "lbProbeID": "[concat(variables('lbID'),'/probes/tcpProbe')]"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[parameters('storageAccountName')]",
+      "apiVersion": "2015-05-01-preview",
```

Comment:
good idea to make api version a variable to make it easier to move to newer api versions. Also use 2015-06-15 api version